### PR TITLE
config: add config for `tidb_enforce_mpp` and remove the global scope of it.

### PR DESCRIPTION
**Pull request by LittleFall**
### What problem does this PR solve?

Problem Summary: need instance scope of var

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed: add config for `tidb_enforce_mpp` and remove the global scope of it.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch: already cherry-picked.

### Check List 

Tests 

- Integration test
- Manual test (add detailed scripts or steps below)
```
1. select @@tidb_enforce_mpp;
result=false
2. add config `performance.enforce-mpp=true`, restart server.
3. select @@tidb_enforce_mpp;
result=true
```

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

- No release note.

### CODE REVIEW

**Review by hanfei1991 — APPROVED** [review]

**Review by lidezhu — APPROVED** [review]

**Review by lidezhu — APPROVED** [review]

**Review by leiysky — APPROVED** [review]

File: sessionctx/variable/sysvar.go [hunk 1]
```diff
@@ -840,7 +840,7 @@ var defaultSysVars = []*SysVar{
 		s.allowMPPExecution = TiDBOptOn(val)
 		return nil
 	}},
-	{Scope: ScopeGlobal | ScopeSession, Name: TiDBEnforceMPPExecution, Type: TypeBool, Value: BoolToOnOff(DefTiDBEnforceMPPExecution), Validation: func(vars *SessionVars, normalizedValue string, originalValue string, scope ScopeFlag) (string, error) {
```
Comment (lzmhhh123) [inline sessionctx/variable/sysvar.go:843]: Is the origin global scope involved in any release version?

Comment (lzmhhh123) [reply sessionctx/variable/sysvar.go:843]: If so, we should remove it in `mysql.global_variables`.

Comment (LittleFall) [reply sessionctx/variable/sysvar.go:843]: No, it is imported in https://github.com/pingcap/tidb/pull/25302 this morning.

Comment (lzmhhh123) [reply sessionctx/variable/sysvar.go:843]: Okay.

**Pull request merged**